admin/class-customizer.php

```diff
@@ -106,149 +125,105 @@ public function breadcrumbs_blog_remove_active_cb() {
-		);
+		$this->add_setting_and_control( $index, $control_args );
 	}
 
 	/**
 	 * Adds the breadcrumbs search prefix text field
 	 */
 	private function breadcrumbs_searchprefix_setting() {
-		$this->wp_customize->add_setting(
-			'wpseo_titles[breadcrumbs-searchprefix]', array(
-				'default'   => '',
-				'type'      => 'option',
-				'transport' => 'refresh',
-			)
+		$index        = 'breadcrumbs-searchprefix';
+		$control_args = array(
+			'label' => __( 'Prefix for search result pages:', 'wordpress-seo' ),
 		);
 
-		$this->wp_customize->add_control(
-			new WP_Customize_Control(
-				$this->wp_customize, 'wpseo-breadcrumbs-searchprefix', array(
-					'label'    => __( 'Prefix for search result pages:', 'wordpress-seo' ),
-					'type'     => 'text',
-					'section'  => 'wpseo_breadcrumbs_customizer_section',
-					'settings' => 'wpseo_titles[breadcrumbs-searchprefix]',
-					'context'  => '',
-				)
-			)
-		);
+		$this->add_setting_and_control( $index, $control_args );
 	}
 
 	/**
 	 * Adds the breadcrumb 404 prefix text field
 	 */
 	private function breadcrumbs_404_setting() {
-		$this->wp_customize->add_setting(
-			'wpseo_titles[breadcrumbs-404crumb]', array(
-				'default'   => '',
-				'type'      => 'option',
-				'transport' => 'refresh',
-			)
+		$index        = 'breadcrumbs-404crumb';
+		$control_args = array(
+			'label' => __( 'Breadcrumb for 404 pages:', 'wordpress-seo' ),
 		);
 
-		$this->wp_customize->add_control(
-			new WP_Customize_Control(
-				$this->wp_customize, 'wpseo-breadcrumbs-404crumb', array(
-					'label'    => __( 'Breadcrumb for 404 pages:', 'wordpress-seo' ),
-					'type'     => 'text',
-					'section'  => 'wpseo_breadcrumbs_customizer_section',
-					'settings' => 'wpseo_titles[breadcrumbs-404crumb]',
-					'context'  => '',
-				)
-			)
-		);
+		$this->add_setting_and_control( $index, $control_args );
+	}
+
+	/**
+	 * Adds the customizer setting and control.
+	 *
+	 * @param string $index         Array key index to use for the customizer setting.
+	 * @param array  $control_args  Customizer control object arguments.
+	 *                              Only those different from the default need to be passed.
+	 * @param string $id            Optional. Customizer control object ID.
+	 *                              Will default to 'wpseo-' . $index.
+	 * @param array  $settings_args Optional. Customizer setting arguments.
+	 *                              Only those different from the default need to be passed.
+	 */
+	private function add_setting_and_control( $index, $control_args, $id = null, $settings_args = array() ) {
+		$setting                  = sprintf( $this->setting_template, $index );
+		$control_args             = array_merge( $this->default_control_args, $control_args );
+		$control_args['settings'] = $setting;
+
+		if ( empty( $settings_args ) ) {
+			$settings_args = $this->default_setting_args;
+		}
+		else {
+			$settings_args = array_merge( $this->default_setting_args, $settings_args );
```

Comment:
In the case that `$settings_args` is empty, you end up merging the `$default_setting_args` property with the `$default_setting_args` property. Wouldn't always merging in the empty `$settings_args` array, work?

Reply:
> In the case that $settings_args is empty, you end up merging the $default_setting_args property with the $default_setting_args property.

No, that's the whole point of doing the `if ( empty() )`, to prevent doing that merge when it's not needed.

> Wouldn't always merging in the empty $settings_args array, work?

I originally had that, but `array_merge()` is actually an expensive operation in terms of resource use in PHP, so doing it when not necessary creates overhead and - so far - none of the calling methods actually pass `$settings_args`, which is why I added the `if/else`.

I could rename the parameter or the local variable, to something else to get rid of the `if/else` if you like, along these lines:
```php
$value = $x;
if ( ! empty( $y ) ) {
    $value = array_merge( $value, $y );
}
```

Reply:
Done ;-)